temporal-testing/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -1517,50 +1517,31 @@ private static State failActivityTask(
 
   private static State failActivityTaskByTaskToken(
       RequestContext ctx, ActivityTaskData data, RespondActivityTaskFailedRequest request) {
-    if (!request.getFailure().hasApplicationFailureInfo()) {
-      throw new IllegalArgumentException("application failure expected: " + request.getFailure());
-    }
-    Failure failure = request.getFailure();
-    RetryState retryState = attemptActivityRetry(ctx, Optional.of(failure), data);
-    if (retryState == RetryState.RETRY_STATE_IN_PROGRESS) {
-      return INITIATED;
-    }
-    data.startedEventId = ctx.addEvent(data.startedEvent);
-    ActivityTaskFailedEventAttributes.Builder a =
-        ActivityTaskFailedEventAttributes.newBuilder()
-            .setIdentity(request.getIdentity())
-            .setScheduledEventId(data.scheduledEventId)
-            .setFailure(request.getFailure())
-            .setRetryState(retryState)
-            .setIdentity(request.getIdentity())
-            .setStartedEventId(data.startedEventId);
-    HistoryEvent event =
-        HistoryEvent.newBuilder()
-            .setEventType(EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED)
-            .setActivityTaskFailedEventAttributes(a)
-            .build();
-    ctx.addEvent(event);
-    return FAILED;
+    return getState(ctx, data, request.getFailure(), request.getIdentity());
   }
 
   private static State failActivityTaskById(
       RequestContext ctx, ActivityTaskData data, RespondActivityTaskFailedByIdRequest request) {
-    if (!request.getFailure().hasApplicationFailureInfo()) {
-      throw new IllegalArgumentException("application failure expected: " + request.getFailure());
+    return getState(ctx, data, request.getFailure(), request.getIdentity());
+  }
+
+  private static State getState(
```

Comment:
This method is structured incorrectly.
It's not a getter (getters don't mutate their arguments and ideally don't have arguments) and the name getState doesn't represent its function (the main function here is creation and publishing of HistoryEvent).

Instead of separating `getState` method this way, here it's better to:
1. Remove one of duplicates. `failActivityTaskByTaskToken` or `failActivityTaskById`. Rename the left one into `failActivityTask`
2. If you want to improve readability of this method, you could separate this piece:
```
    data.startedEventId = ctx.addEvent(data.startedEvent);
     ActivityTaskFailedEventAttributes.Builder a =
         ActivityTaskFailedEventAttributes.newBuilder()
             .setIdentity(request.getIdentity())
             .setScheduledEventId(data.scheduledEventId)
             .setFailure(request.getFailure())
             .setRetryState(retryState)
             .setIdentity(request.getIdentity())
             .setStartedEventId(data.startedEventId);
     HistoryEvent event =
         HistoryEvent.newBuilder()
             .setEventType(EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED)
             .setActivityTaskFailedEventAttributes(a)
             .build();
     ctx.addEvent(event);
```
into a separate method with a meaningful name.